app/jobs/plexer_job.rb

```diff
@@ -1,62 +1,54 @@
+# Record zip part metadata info in DB.
 # Split message out to all necessary endpoints.
 # For example:
-#   Endpoint1Delivery.perform_later(druid, version)
-#   Endpoint2Delivery.perform_later(druid, version)
+#   Endpoint1Delivery.perform_later(druid, version, part_s3_key)
+#   Endpoint2Delivery.perform_later(druid, version, part_s3_key)
 #   ...
 #
 # Do not assume we can just get metadata from (the DruidVersionZip) zip.
-# Jobs are not run at the same time or on the same system, so the info will not match.
+# Jobs are not run at the same time or on the same system, so the info may not match.
 # Therefore, we receive the info passed by the process that was there when the file was created.
-class PlexerJob < DruidVersionJobBase
+class PlexerJob < ZipPartJobBase
   queue_as :zips_made
 
-  before_enqueue { |job| job.zip_info_check!(job.arguments.third) }
+  before_enqueue { |job| job.zip_info_check!(job.arguments.fourth) }
 
   # @param [String] druid
   # @param [Integer] version
+  # @param [String] part_s3_key, e.g. 'ab/123/cd/4567/ab123cd4567.v0001.z03'
   # @param [Hash<Symbol => String>] metadata Zip info
+  # @option metadata [Integer] :parts_count
   # @option metadata [Integer] :size
   # @option metadata [String] :checksum_md5
+  # @option metadata [String] :suffix
   # @option metadata [String] :zip_cmd
   # @option metadata [String] :zip_version
-  def perform(druid, version, metadata)
-    ApplicationRecord.transaction do
-      update_size(metadata[:size])
-      pcs.each do |pc|
-        pc.zip_checksums.find_or_create_by!(
-          md5: metadata[:checksum_md5],
-          create_info: metadata.slice(:zip_cmd, :zip_version).to_s
-        )
-      end
+  def perform(druid, version, part_s3_key, metadata)
+    apcs.each do |apc|
+      apc.archive_preserved_copy_parts.find_or_create_by(
+        create_info: metadata.slice(:zip_cmd, :zip_version).to_s,
+        md5: metadata[:checksum_md5],
+        parts_count: metadata[:parts_count],
+        size: metadata[:size],
+        suffix: File.extname(part_s3_key)
+      ) { |part| part.status = 'unreplicated' }
+      apc.save!
     end
-    targets(pcs.pluck(:endpoint_id)).each { |worker| worker.perform_later(druid, version, metadata) }
-  end
-
-  # @return [PreservedCopy::ActiveRecord_Relation]
-  def pcs
-    @pcs ||= PreservedCopy.by_druid(zip.druid.id).for_archive_endpoints.where(version: zip.version)
+    deliverers.each { |worker| worker.perform_later(druid, version, part_s3_key, metadata) }
   end
 
-  # @param [Array<Integer>] endpoint_ids
-  # @return [Array<Class>] Endpoint delivery classes to be targeted
-  def targets(endpoint_ids)
-    Endpoint
-      .where(id: endpoint_ids)
-      .group(:delivery_class, :id)
-      .pluck(:delivery_class, :id)
-      .map do |pair|
-        Rails.logger.error("Archive Endpoint (id: #{pair.second}) has no delivery_class") unless pair.first
-        pair.first
-      end.compact
+  # @return [ArchivePreservedCopy]
```

Comment:
one of these:  http://api.rubyonrails.org/classes/ActiveRecord/Relation.html

by ruby's duck-typing philosophy, the method comment is honest, since the `Relation` will behave as if it were an `ArchivePreservedCopy` array for most purposes, but i agree it might be nice to be more specific (i'd also be ok w/ just pulling this particular comment).